python/example_code/rekognition/custom_labels_csv_to_manifest.py

```diff
@@ -0,0 +1,200 @@
+import json
+
+"""
+Purpose
+Amazon Rekognition Custom Labels model example used in the service documentation:
+Shows how to create an image-level (classification) manifest file from a CSV file.
+You can specify multiple image level labels per image.
+CSV file format is
+image,label,label,..
+If necessary, use the bucket argument to specify the S3 bucket folder for the images.
+https://docs.aws.amazon.com/rekognition/latest/customlabels-dg/md-gt-cl-transform.html
+"""
+
+logger = logging.getLogger(__name__)
+
+def check_duplicates(csv_file, deduplicated_file, duplicates_file):
+    """
+    Checks for duplicate images in a CSV file. If duplicate images
+    are found, deduplicated_file is the deduplicated CSV file - only the first
+    occurence of a duplicate is recorded. Other duplicates are recorded in duplicates_file.
+    :param csv_file: The source CSV file
+    :param deduplicated_file: The deduplicated CSV file to create. If no duplicates are found
+    this file is removed.
+    :param duplicates_file: The duplicate images CSV file to create. If no duplicates are found
+    this file is removed.
+    :return: True if duplicates are found, otherwise false.
+    """
+
+    logger.info("Deduplicating %s", csv_file)
+
+    duplicates_found = False
+
+    #Find duplicates
+    with open(csv_file, 'r', newline='', encoding="UTF-8") as f,\
+        open(deduplicated_file, 'w', encoding="UTF-8") as dedup,\
+        open(duplicates_file, 'w', encoding="UTF-8") as duplicates:
+
+        reader = csv.reader(f, delimiter=',')
+        dedup_writer = csv.writer(dedup)
+        duplicates_writer = csv.writer(duplicates)
+
+        entries = set()
+        for row in reader:
+            #Skip empty lines
+            if not ''.join(row).strip():
+                continue
+
+            key = row[0]
+            if key not in entries:
+                dedup_writer.writerow(row)
+                entries.add(key)
+            else:
+                duplicates_writer.writerow(row)
+                duplicates_found = True
+
+    if duplicates_found:
+        logger.info("Duplicates found check %s", duplicates_file)
+
+    else:
+        os.remove(duplicates_file)
+        os.remove(deduplicated_file)
+
+    return duplicates_found
+
+
+def create_manifest_file(csv_file, manifest_file, s3_path):
+    """
+    Reads a CSV file and creates a Custom Labels classification manifest file.
+    :param csv_file: The source CSV file.
+    :param manifest_file: The name of the manifest file to create.
+    :param s3_path: The S3 path to the folder that contains the images.
+    """
+    logger.info("Processing CSV file %s", csv_file)
+
+    image_count = 0
+    label_count = 0
+
+    with open(csv_file, newline='', encoding="UTF-8") as csvfile,\
+        open(manifest_file, "w", encoding="UTF-8") as output_file:
+
+        image_classifications = csv.reader(
+            csvfile, delimiter=',', quotechar='|')
+
+        # process each row (image) in CSV file
```

Comment:
Process each row (image) in the CSV file.

Reply:
Fixed.